keepassxc_proxy_client/protocol.py

```diff
@@ -160,8 +160,29 @@ def get_logins(self, url):
         else:
             return response["entries"]
 
+    def get_database_groups(self):
+        msg = {
+            "action": "get-database-groups",
+        }
+
+        self.send_encrypted_message(msg)
+        response = self.get_encrypted_response()
+        return response
+
     def get_unencrypted_response(self):
-        return json.loads(self.socket.recv(4096).decode("utf-8"))
+        data = []
+        while True:
+            try:
+                new_data = self.socket.recv(4096)
+                if new_data:
+                    data.append(new_data.decode('utf-8'))
+                else:
+                    break
+                if len(new_data) < 4096:
+                    break
+            except:
+                pass
```

Comment:
If the socket gets closed prematurely this will send the function into an endless loop  since the `recv` will always throw an exception and cause a new iteration

Reply:
Good point! Can't see a good point for swallowing the exceptions in the recv() loop. The fixup commit removes it.

AFAIR UNIX datagram sockets don't throw on connection lost, as it is not connection oriented. I tried the following, no exception thrown:

Server:
```bash
socat unix-listen:/tmp/testunix stdin
```

Client
```python
import socket
sock  = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
sock.connect("/tmp/testunix")
sock.recv(4096)
```